client/homebrew/brewRenderer/toolBar/toolBar.jsx

```diff
@@ -37,10 +39,67 @@ const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages })=>{
 		setPageNum(pageNumber);
 	};
 
+
+	const calculateZoom = (mode)=>{
+		const iframe = document.getElementById('BrewRenderer');
+		const iframeWidth = iframe.getBoundingClientRect().width;
+		const iframeHeight = iframe.getBoundingClientRect().height;
+		const pages = iframe.contentWindow.document.getElementsByClassName('page');
+
+		let desiredZoom = 0;
+
+		if(mode == 'fill'){
+			// find widest page, in case pages are different widths, so that the zoom is adapted to not cut the widest page off screen.
+			let widestPage = 0;
+			[...pages].forEach((page)=>{
+				const width = page.offsetWidth;
+				if(width > widestPage)
+					widestPage = width;
+			});
```

Comment:
Seems like an opportunity for a `reduce`. What about something like this?

```suggestion
			const widestPage = [...pages].reduce((maxWidth, page) => Math.max(maxWidth, page.offsetWidth), 0);
```

Reply:
wow that maxBy is good.  done.